compiler/src/iree/compiler/Codegen/Common/MaterializeEncodingPatterns.cpp

```diff
@@ -278,13 +307,51 @@ static FailureOr<Operation *> lowerGenericOpWithEncoding(
       auto dimPos = cast<AffineDimExpr>(inputMap.getResult(pos)).getPosition();
       for (auto [tileIdx, outDim] : llvm::enumerate(outInnerDimsPos)) {
         if (dimPos == outDim) {
-          packedResultDims.push_back(outputMap.getNumDims() + tileIdx);
+          // In case of an encoding with swizzle, an expanded set of dimensions
+          // needs to be appended as specified by the swizzle's `expandedShape`
+          // field. Note that the dimension index should be offset by the
+          // calculated output starting offset as every dimension is now
+          // transformed into an expanded sequence of indices and the correct
+          // dimension index is:
+          //   outOffsetForDimsPos[tileIdx] + innerIndex
+          if (materializeEncodingInfo.swizzle.has_value()) {
+            assert(idx <= materializeEncodingInfo.swizzle->expandShape.size() &&
+                   "`innerDimsPos` index should not exceed the swizzle's "
+                   "`expandShape` size");
```

Comment:
I think this should be checking `<`
```suggestion
            assert(idx < materializeEncodingInfo.swizzle->expandShape.size() &&
                   "`innerDimsPos` index should not exceed the swizzle's "
                   "`expandShape` size");
```

Reply:
Yes, thanks.